### PR TITLE
Script to update locales in CI

### PR DESCRIPTION
**Pull request by nicksellen**
#### Proposed Changes

* runs on master branch in CI
* fixes weblate JSON issues (see #1487)
* extracts new translations keys (see #1486)
* commits and pushes any changes to git repo
* then runs build/deploy as usual

Uses a robot user I created (https://github.com/trustroots-ci) to do the commits.

The workflow for incorporate the weblate translations will then be to just merge the weblate PR whenever we want (even with it's JSON mistakes), and then this would run and fix it all.

#### Testing Instructions

* hard to test! once reviewed visually, I will merge to master and then fiddle until it's working :)

Fixes https://github.com/Trustroots/trustroots/issues/1487 and https://github.com/Trustroots/trustroots/issues/1486

### CODE REVIEW

File: scripts/ci/fix-locales-and-push-changes [hunk 1]
```diff
@@ -0,0 +1,39 @@
+#!/bin/sh
+
+fix() {
+  ./scripts/ci/fix-weblate public/locales/*/*.json
+  npx prettier --write public/locales/*/*.json
```
Comment (simison) [inline scripts/ci/fix-locales-and-push-changes:5]: Doesn't this happen in git commit?

File: scripts/ci/update-locales [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+[ci skip]
+EOF
+  fi
+}
+
+push_changes() {
+  if [ ! -z "$TRAVIS_BRANCH" ]; then
+    echo "Pushing changes!"
+    git config --global user.email "trustroots-ci@nicksellen.co.uk" # TODO: move this to @trustroots.org account...
```
[the remark this answers is not in the document]
Comment (nicksellen) [reply scripts/ci/update-locales:34]: Is _is_ a real GitHub account, I created it, GitHub call them _machine users_ (see https://developer.github.com/v3/guides/managing-deploy-keys/#machine-users for them in another context). And in this case it's to be able to create a personal access token that only has access to what it needs, if I create one in my own account it has _way_ too much access...

File: scripts/ci/fix-locales-and-push-changes (hunk 1 above)
Comment (nicksellen) [reply scripts/ci/fix-locales-and-push-changes:5]: Yes, good point, although it doesn't actually work when I tried it here... I just managed to commit:

```json
{
  "Polish":           "Polština",
     "foo": "bar"
}
```
locally... and no fixing occured :/

... digdigdig, `.eslintignore` says to ignore `public`...  we re-use the ignore file from eslint for prettier, _but_ they have different syntaxes/semantics (a bit)...

- https://eslint.org/docs/user-guide/configuring#eslintignore
- https://git-scm.com/docs/gitignore#_pattern_format

working on it...

Comment (nicksellen) [reply scripts/ci/fix-locales-and-push-changes]: Fixed in the last few commits... oops also seeing I set my local git credentials to the Trustroots CI user :)

**Comment by nicksellen** [comment]
A big failure so far:
- it runs when it's not even on the master branch
- we can't push to master
- ... it _tries_ to push to master (I guess `$TRAVIS_BRANCH` is not what I expected...)

> $ [ "$TRAVIS_BRANCH" = "master" ] && ./scripts/ci/update-locales
> trustroots@0.3.3 i18n:extract /home/travis/build/Trustroots/trustroots
> babel --config-file ./babel.config.i18n.js -o /dev/null 'modules/**/*.js'
> [detached HEAD 9e18402] [CI] Extract locales
> 40 files changed, 3652 insertions(+), 20 deletions(-)
> Pushing changes!
> remote: error: GH006: Protected branch update failed for refs/heads/master.        
> remote: error: Required status check "continuous-integration/travis-ci" is expected. At least 1 approving review is required by reviewers with write access.        
> To https://github.com/Trustroots/trustroots.git
>  ! [remote rejected] HEAD -> master (protected branch hook declined)
> error: failed to push some refs to 'https://trustroots-ci:[secure]@github.com/Trustroots/trustroots.git'

**Comment by nicksellen** [comment]
Ok, that should all be fixed now, but won't really know until it's merged. So any more reviews welcome :)

File: .eslintignore [hunk 3]
```diff
@@ -3,7 +3,8 @@
 bin/admin
 deploy
 migrations/archived
-public
+public/*
+!public/locales
```
Comment (simison) [inline .eslintignore:7]: `!public/**/*.json` could work too, I see there's another json file that we might want to Prettify.

Not a blocker.

**Comment by simison** [comment]
Should we mergee this first? https://github.com/Trustroots/trustroots/pull/1373

**Comment by nicksellen** [comment]
Ok, gunna go for it after the CI finishes... no need to merge #1373 before. It will test the flow better to do it after.

**Comment by nicksellen** [comment]
.... ok it's not very happy, as committing/pushing to master is VERY NAUGHTY and I can't seem to make an exemption. Maybe if I make the bot user an admin...

**Comment by nicksellen** [comment]
Having the CI user as an admin user worked, it's a bit overprivileged but hey ho.

> I wonder if we'd be better off to use GitHub Actions — could simplify this part?

GitHub actions is basically just another CI provider, just built in. Maybe the user rights are easier, but for everything else seems more painful having two CI providers at the same time.

**Pull request merged**